src/components/query-viewer/query-viewer.tsx

```diff
@@ -287,12 +358,346 @@ export class QueryViewer implements GxComponent {
+    event: CustomEvent<PivotTableExpandCollapse>
+  ) {
+    const eventData = itemExpandCollapsePivotTableEvent(
+      (event as any).parameter.Queryviewer,
+      (event as any).parameter.Data,
+      (event as any).parameter.IsCollapse
+    );
+
+    if ((event as any).parameter.IsCollapse) {
+      this.itemCollapse.emit(eventData);
+    } else {
+      this.itemExpand.emit(eventData);
+    }
+  }
+
+  @Listen("PivotTableOnPageChangeEvent")
+  handlePivotTableOnPageChangeEvent(event: CustomEvent<PivotTablePageChange>) {
+    let eventData;
+
+    switch ((event as any).parameter.Navigation) {
+      case PivotTableNavigation.OnFirstPage:
+        eventData = this.pivotRenderRef.firstPage();
+        this.OnFirstPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnLastPage:
+        eventData = this.pivotRenderRef.lastPage();
+        this.OnLastPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnNextPage:
+        eventData = this.pivotRenderRef.nextPage();
+        this.OnNextPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnPreviousPage:
+        eventData = this.pivotRenderRef.previousPage();
+        this.OnPreviousPage.emit(eventData);
+        break;
+    }
+  }
+
+  @Listen("PivotTableOnFilterChangedEvent")
+  handlePivotTableOnFilterChangedEvent(
+    event: CustomEvent<PivotTableFilterChanged>
+  ) {
+    console.log(event.detail);
+    // const eventData = onFilterChangedPivotTableEvent(
+    //   (event as any).parameter.QueryviewerId,
+    //   (event as any).parameter.FilterChangedData
+    // );
+    // this.filterChanged.emit(eventData);
+  }
+
+  /** User Events for Table*/
+
+  // ToDo: implement this
+  @Listen("TableOnItemClickEvent")
+  handleTableOnItemClickEvent(event: CustomEvent<TableItemClick>) {
+    console.log(event.detail);
+  }
+  // ToDo: implement this
+  @Listen("TableOnPageChangeEvent")
+  handleTableOnDragundDropEvent(event: CustomEvent<TablePageChange>) {
+    console.log(event.detail);
+  }
+  // ToDo: implement this
+  @Listen("TableOnFilterChangedEvent")
+  handleTableOnFilterChangedEvent(event: CustomEvent<TableFilterChanged>) {
+    console.log(event.detail);
+  }
+
+  /** GX User Events */
+
+  /**
+   * Returns an XML on a string variable containing all the data for the attributes loaded in the Pivot Table.
+   */
+  @Method()
+  async getData() {
+    if (!this.controller) {
+      return;
+    }
+    switch (this.OutputType) {
+      case QueryViewerOutputType.Card:
+        // ToDo: implement this method to the output card
+        return null;
+      case QueryViewerOutputType.Chart:
+        // ToDo: implement this method to the output chart
+        return null;
+      case QueryViewerOutputType.Map:
+        // ToDo: implement this method to the output map
+        return null;
+      default: // PivotTable and Table
+        const serverData = this.pivotRenderRef.getPivottableDataSyncXml;
```

Comment:
I think we are missing the following
```suggestion
        const serverData = await this.pivotRenderRef.getPivottableDataSyncXml();
```

Reply:
Although it seems strange that we have to call 2 methods of the pivot-render to resolve the value, instead of having a single method for this case in the pivot-render